code-challenges.test.js

```diff
@@ -22,25 +38,168 @@ const people = [
+      // use join(" ") to join the first name and last name into one string
+      // using string interpolation add the rest of the sentence using dot notation to access value of occupation key 
+
+
+const upName = (array) => {
+ return array.map(value => value.name.split(" ").map(fullName => fullName.charAt(0).toUpperCase() + fullName.slice(1)).join(" ") + ` is ${value.occupation}.`
+
+  )
+}
+
+
+// I think the reason why previous code didn't work the first time was because I was storing the first map iteration in a variable and so when I called to that variable - it was still an array. This meant that I wasn't able to use any string methods on my return because I couldn't acces the string inside. By doing the code execution on one line, array.map(value => value.name.split(" ") allows me to access the values and split them into separate strings and then the function to the right of the arrow (fullName.charAt(0).toUpperCase() + fullName.slice(1)).join(" ") + ` is ${value.occupation}.`) allows me to perform iterations and use the different string methods on the values that have already been accessed. 
+
+
+// green of green-red refactor: 
+  // PASS  ./code-challenges.test.js
+  // upName
+
 
 // --------------------2) Create a function that takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3.
 
 // a) Create a test with an expect statement using the variables provided.
 
+// Pseudocode: 
+  // describe a method that lists the name of function called remainderArr 
+  // nested within the describe block, have a test/it method that describes what the function does "It will take an a mixed data array and return an array of only the remainder of the numbers when divided by 3"
+  // nested in the test block, have an expect method that is calling the function remainderArr have an expect method followed by the .toEqual() method matcher that will check the expected output of the function 
+
+describe("remainderArr", () => {
+  it("It will take an a mixed data array and return an array of only the remainder of the numbers when divided by 3", () => {
+    expect(remainderArr(hodgepodge1)).toEqual([ 2, 0, -1, 0 ]),
+    expect(remainderArr(hodgepodge2)).toEqual([2, 1, -1])
+
+  })
+})
+
+// red of red-green refactor: 
+  //  FAIL  ./code-challenges.test.js
+  // ReferenceError: remainderArr is not defined
+
+
+
 const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
 // Expected output: [ 2, 0, -1, 0 ]
 const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
 // Expected output: [ 2, 1, -1 ]
 
 // b) Create the function that makes the test pass.
 
+// Pseudocode: 
+  // declare a function named remainderArr 
+  // set a new variable to hold: 
+    // iterate through array using filter method to locate only numbers (typeof?)
+  // return the variable set and use .map method to iterate through array of only numbers and use modulo operator % 3 so we can return the remainder 
+
```

Comment:
next steps will be the refactor. Make filter do all the work. hint you can return a condition with the logical operator &&